### ISSUE TITLE
NVDA Does Not Announce OneDrive Folder Names in Explorer Navigation Pane

### ISSUE THREAD

**Issue by keyang556** [nvaccess/nvda#20572]
### Brief summary

NVDA does not correctly announce OneDrive folder names in the Windows 11 File Explorer navigation pane or in standard Windows "Save As" dialogs. Instead of speaking the folder name (e.g., "OneDrive," "Documents," "Desktop"), NVDA announces only generic tree view information such as: "tree view item expanded 3 of 11 level 1" or "tree view item collapsed 7 of 11 level 2"
Windows Narrator correctly announces the same OneDrive items and folder names.

### Steps to reproduce

1. Open file explorer.
2. Browse folders, tree view.
3. Press up or down, the onedrive folder not report the folder name.

### Actual behavior

NVDA announces only generic tree view information such as: "tree view item expanded 3 of 11 level 1" or "tree view item collapsed 7 of 11 level 2"

### Expected behavior

read the folder name

### NVDA logs, crash dumps and other attachments

[NVDA.Log.txt](https://github.com/user-attachments/files/30438978/NVDA.Log.txt)

[nvda.log](https://github.com/user-attachments/files/30438992/nvda.log)

### NVDA type

installed copy

### NVDA version

alpha-57160,fd46cdc6 (2026.3.0.57160)

### Have you tried any other versions of NVDA? If so, please report their behaviors.

2026.1.1 same problem

### Windows version

Windows 11 Dev 29617.1000  (someone says 10.0.26200.8893 same problem)

### Name and version of other software in use when reproducing the issue

Onedriv26.129.0706.0003e 

### Other information about your system

NO

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I have restarted NVDA with add-ons disabled and the issue still occurs

### Does the issue still occur after you run the System Accessibility Repair Tool (COM Registration Fixing Tool in older versions) in NVDA's tools menu?

I have run the System Accessibility Repair Tool and the issue still occurs

**Comment by keyang556** [comment]
Related #20570

**Comment by josephsl** [comment]
Hi,

I cannot reproduce this. I opened File Explorer, pressed Shift+Tab to move to folders tree view, expanded my OneDrive folder, and items in there are read by NVDA.

Setup:

* NVDA version: alpha-57160
* Windows version: Windows 11 26H2 Experimental (build 26300.8935)

Thanks.

**Comment by josephsl** [comment]
Hi,

On my system, OneDrive entries in the folder tree view are IAccessible objects. Could you post the developer info for the item not spoken by NVDA? That way we can figure out if UIA handling is indeed a factor (we cannot really say this is a UIA problem until developer info for the unspoken item is obtained).

Thanks.

**Comment by keyang556** [comment]
>Hi,
On my system, OneDrive entries in the folder tree view are IAccessible objects. Could you post the developer info for the item not spoken by NVDA? That way we can figure out if UIA handling is indeed a factor (we cannot really say this is a UIA problem until developer info for the unspoken item is obtained).
Thanks.

How to get the developer info? Thanks.

**Comment by keyang556** [comment]
Narrator report the folder name correctly.
This issue was started about one or two months ago, maybe after I sign out and sign in onedrive again(not sure), or onedrive version.

**Comment by tareh7z** [comment]
> > Hi,
> > On my system, OneDrive entries in the folder tree view are IAccessible objects. Could you post the developer info for the item not spoken by NVDA? That way we can figure out if UIA handling is indeed a factor (we cannot really say this is a UIA problem until developer info for the unspoken item is obtained).
> > Thanks.
> 
> How to get the developer info? Thanks.

Hi @keyang556!

First, set the focus to the object you want to get information about, then press `NVDA+F1`. This will open the Log Viewer. Your focus should be on the line just above the one that says "Developer info for navigator object:". Copy everything below that line and paste it here.

Thanks.

**Comment by josephsl** [comment]
Hi,
When you locate the unspoken item, press NVDA+F1. The NVDA log opens. Then copy and paste the part of the log that begins with "developer info" and to the end.

Thanks.

**Comment by keyang556** [comment]
Developer info for navigator object:
name: None
role: TREEVIEWITEM
processID: 12168
roleText: None
states: COLLAPSED, SELECTABLE, FOCUSED, FOCUSABLE, SELECTED
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.Dynamic_TreeViewItemOutlineItemIAccessible object at 0x00000269FADBF110>
Python class mro: (<class 'NVDAObjects.Dynamic_TreeViewItemOutlineItemIAccessible'>, <class 'NVDAObjects.IAccessible.sysTreeView32.TreeViewItem'>, <class 'NVDAObjects.IAccessible.OutlineItem'>, <class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <class 'garbageHandler.TrackedObject'>, <class 'object'>)
description: None
location: RectLTWH(left=367, top=770, width=40, height=48)
value: None
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
appModule: AppModule(explorer, appName='explorer', processID=12168)
appModule.productName: 'Microsoft® Windows® Operating System'
appModule.productVersion: '10.0.29634.1000'
appModule.helperLocalBindingHandle: None
appModule.appArchitecture: 'AMD64'
windowHandle: 11141886
windowClassName: 'SysTreeView32'
windowControlID: 100
windowStyle: 1445040685
extendedWindowStyle: 0
windowThreadID: 21888
windowText: '瀏覽窗格'
displayText: ''
IAccessibleObject: <POINTER(IAccessible) ptr=0xfb088fe3a0 at 269fbf46250>
IAccessibleChildID: 13
IAccessible event parameters: windowHandle=11141886, objectID=-4, childID=13
IAccessible accName: exception: (-2147023728, '元素找不到。', (None, None, None, 0, None))
IAccessible accRole: ROLE_SYSTEM_OUTLINEITEM
IAccessible accState: STATE_SYSTEM_SELECTED, STATE_SYSTEM_FOCUSED, STATE_SYSTEM_COLLAPSED, STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_SELECTABLE, STATE_SYSTEM_VALID (3146758)
IAccessible accDescription: None
IAccessible accValue: '2'

**Comment by seanbudd** [comment]
Can anyone else reproduce this issue?

**Comment by keyang556** [comment]
fixed
